SF iOS/SF iOS/Feed/SwiftUI/Data.swift

```diff
@@ -0,0 +1,140 @@
+//
+//  Data.swift
+//  Coffup
+//
+//  Created by Roderic Campbell on 6/8/19.
+//
+
+import UIKit
+import SwiftUI
+import CoreLocation
+import Combine
+
+struct WrappedEvent /*, BindableObject*/ {
+//    var didChange = (NotificationCenter.default.publisher(for: "eventChange") as AnyObject).receive(on: RunLoop.main)
+//    var event: Event
+    let name: String
+    let eventID: String
+    let startDate: String
+    let venue: Venue
+    var locationCoordinate: CLLocationCoordinate2D {
+        CLLocationCoordinate2D(
+            latitude: venue.location.latitude,
+            longitude: venue.location.longitude)
+    }
+    
+    enum CodingKeys: String, CodingKey {
+        case eventID = "id"
+        case name = "name"
+        case venue = "venue"
+        case startDate = "start_at"
+    }
+}
+
+struct Venue: Hashable, Codable {
+    let name: String
+    let location: Location
+    enum CodingKeys: String, CodingKey {
+        case name = "name"
+        case location = "location"
+    }
+}
+
+struct Location: Hashable, Codable {
+    let address: String
+    let latitude: Double
+    let longitude: Double
+    
+    enum CodingKeys: String, CodingKey {
+        case address = "formatted_address"
+        case latitude = "latitude"
+        case longitude = "longitude"
+    }
+}
+
+extension WrappedEvent: Decodable {
+}
+
+
+let eventData: [WrappedEvent] = load("eventData.json")
+
+func load<T: Decodable>(_ filename: String, as type: T.Type = T.self) -> T {
+    let data: Data
+    print("this is good")
+    guard let file = Bundle.main.url(forResource: filename, withExtension: nil)
+        else {
+            fatalError("Couldn't find \(filename) in main bundle.")
+    }
+    
+    do {
+        data = try Data(contentsOf: file)
+    } catch {
+        fatalError("Couldn't load \(filename) from main bundle:\n\(error)")
+    }
+    
+    do {
+        let decoder = JSONDecoder()
+        return try decoder.decode(T.self, from: data)
+    } catch {
+        fatalError("Couldn't parse \(filename) as \(T.self):\n\(error)")
+    }
+}
+
+final class ImageStore {
```

Comment:
Currently never used. FYI I never actually do a network request for data or images.

Reply:
Could we rename T to something more explicit?

Reply:
I _think_ this is how swift defines a generic type in a generic function. I might be saying that wrong. Basically I'm pretty sure it's a reserved word in swift. See: The section near the top for `Generic Functions` at https://docs.swift.org/swift-book/LanguageGuide/Generics.html

> Generic functions can work with any type. Here’s a generic version of the swapTwoInts(_:_:) function from above, called swapTwoValues(_:_:):

And the attached code looks like this:
```
func swapTwoValues<T>(_ a: inout T, _ b: inout T) {
    let temporaryA = a
    a = b
    b = temporaryA
}
```

At any rate, this is actually copy/pasted from the swiftUI samples.